src/utils/AppRouteProvider.jsx

```diff
@@ -10,9 +15,15 @@ const AppRouterContext = React.createContext(null);
 const AppRouteProvider = (props) => {
```

Comment:
could you explain what this `AppRouterProvider` does?, Is it used for all the navigation inside cellenics? or only for some routes

Reply:
`AppRouteProvider` is a component which wraps the application and acts like a middleware for route changes. It intercepts navigation and allow checking of conditions before letting the navigation proceed. Yes, after this PR, all navigation inside Cellenics is done via `AppRouteProvider`.

Reply:
could you add then that description as documentation to the `AppRouteProvider`. Have you checked that no other calls are being made to `router.push`? if there are other calls, we might want to consider this a refactor and have some more tickets to stop using `router.push`. Having both even for some time would induce to error.

Reply:
Yes, I have checked there is no other places where `router.push` is used. I'll add a documentation for the component.

Reply:
Documentation added in 3cd1744